flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/CustomDistributionITCase.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.test.javaApiOperators;
+
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.common.functions.RichMapPartitionFunction;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.test.distribution.CustomDistribution;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.api.java.utils.DataSetUtils;
+import org.apache.flink.types.IntValue;
+import org.apache.flink.util.Collector;
+import org.junit.Test;
+
+
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+
+public class CustomDistributionITCase {
+	
+	@Test
+	public void testPartitionWithDistribution1() throws Exception{
+		/*
+		 * Test the record partitioned rightly with one field according to the customized data distribution
+		 */
+
+		ExecutionEnvironment env = ExecutionEnvironment.createLocalEnvironment();
+
+		DataSet<Tuple3<Integer, Integer, String>> input1 = env.fromElements(
+				new Tuple3<>(1, 1, "Hi"),
+				new Tuple3<>(1, 2, "Hello"),
+				new Tuple3<>(1, 3, "Hello world"),
+				new Tuple3<>(2, 4, "how are you?"),
+				new Tuple3<>(2, 5, "I am fine."),
+				new Tuple3<>(3, 6, "Luke Skywalker"),
+				new Tuple3<>(4, 7, "Comment#1"),
+				new Tuple3<>(4, 8, "Comment#2"),
+				new Tuple3<>(4, 9, "Comment#3"),
+				new Tuple3<>(5, 10, "Comment#4"));
+
+		final IntValue[] keys = new IntValue[3];
+
+		for (int i = 0; i < 3; i++) {
+			keys[i] = new IntValue((i + 1) * 2);
+		}
+
+		final CustomDistribution cd = new CustomDistribution(keys);
+
+		env.setParallelism(3);
+
+		DataSet<Boolean> out1 = DataSetUtils.partitionByRange(input1.map(new MapFunction<Tuple3<Integer, Integer, String>, Tuple2<IntValue, IntValue>>() {
+			@Override
+			public Tuple2<IntValue, IntValue> map(Tuple3<Integer, Integer, String> value) throws Exception {
```

Comment:
This map function is not necessary, if the `DataDistribution` returns `Integer` instead of `IntValue`.